invenio_vocabularies/services/generators.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+#
+
+"""Vocabulary generators."""
+
+from invenio_access import any_user, authenticated_user
+from invenio_records_permissions.generators import Generator
+from invenio_search.engine import dsl
+
+
+class IfTags(Generator):
+    """Generator to filter based on tags.
+
+    This generator will filter records based on the tags field.
+    Optionally, it can be configured to only allow authenticated users.
+    """
+
+    def __init__(self, include=None, exclude=None, only_authenticated=False):
+        """Constructor."""
+        self.include = include or []
+        self.exclude = exclude or []
+        self.only_authenticated = only_authenticated
+
+    def needs(self, **kwargs):
+        """Enabling Needs."""
+        return [authenticated_user] if self.only_authenticated else [any_user]
```

Comment:
Is only authenticated correct here? wasn'it it that only admins should be able to get the unlisted?